merkle/src/lib.rs

```diff
@@ -0,0 +1,23 @@
+// Copyright 2020 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0
+
+#![deny(warnings)]
+#![deny(missing_docs)]
+
+//! mina-merkle crate provides traits and data structure implementations for
+//! in-memory, persistent, maskable and masking mina merkle tree
```

Comment:
A little dive into how all of the maskable and merkle works in the context of mina would be enlightening once we have the details.

Reply:
maskable / masking features are trait only, will add more context in docs when I start implementing them, not in this PR tho.